### DISCUSSION TITLE
Looking for co-maintainers

### DISCUSSION THREAD

**Discussion by natefinch** [magefile/mage#369] [Announcements]
I love working on Mage, but I have three kids and a full time job, and I don't have time to follow up on every PR and issue myself. This needs to be a community project, run by the Go community for the Go community.

There are a few key tenets I have for Mage:
Mage should do the boring parts of wiring up a CLI to dev-focused code for you.
Mage should do what you hope would Just Work™ in the simplest, most obvious way.
Mage should get out of your way so you can write the logic to your dev tool in the clearest and most obvious way possible.
Mage code should be plain old Go code. No special syntax... any go code you want to use, you can.
Mage itself should not have dependencies outside the stdlib and its own libraries.
Mage must remain backwards compatible with scripts written for earlier versions of itself.
Mage must be extremely well tested, since it may be driving major portions of a company's CI and/or dev team.
Mage must be as fast as possible. 
Mage's code must be a good example of idiomatic Go code.

I welcome anyone who is interested in working on features, bug fixes, code reviews etc.  I don't exactly know yet how to structure such things so there's some organization to the chaos, but we'll figure that out together. I imagine starting off as the manager/team lead to set the tone and guide the ship while new folks get their feet under them and learn the codebase.

So, if you're interested, drop a response here, and start replying to issues and reviewing PRs. 

Thanks, y'all.
-Nate

**Comment by sheldonhull** [comment]
@natefinch I'd be up for helping however I can. This project has been a big help to me. 
Still picking up Go, so might not be able to do anything advanced initially, but I'm working on it and vested in something like this as I use it everyday right now :-) Add me to your list of Gopher minions 👋🏻 

I've messed around a bit with GitHub actions so if you don't have a lot of that running I know that stuff pretty decently if say golangci-lint feedback, goreleaser configs, stalebot config, conventional commit/standard setups for PRs and such would be useful. Just some thoughts, in case those "chore" pieces haven't already been setup, let me know if I could help out initially with those too. I bootstrap a lot of my repos with stuff like that.

**Reply by arschles** [reply]
I use mage for so much, including driving the whole [KEDA HTTP Addon](https://github.com/kedacore/http-add-on) project and more. I'm very happy to give what I can back to this project. @natefinch I'll start with reviewing some PRs and commenting on issues like you asked.

**Comment by ladydascalie** [comment]
Hello @natefinch, I'm very much interested in helping.
I've worked with mage both personally, and at work, for quite some time now, so I have vested interest in making sure this amazing project continues to succeed.

I've written Go professionally for the past 5 years and I'm happy to get in touch and let you know more about my credentials if it helps.

Very happy to learn more about the code base and see if I can find ways to make myself useful!

**Comment by thaney071** [comment]
Hello @natefinch 

I just found this project so I haven't used it yet. I am very interested in learning more.  I have been working professionally as a software engineer for 5 years, 3 of which has been mostly in go.  I would love to help out. 

I left a comment on one issue that I would like to help with.

**Comment by viktorvoltaire** [comment]
Hi @natefinch, I would be happy to assist with the project
